src/test/globalSetup.test.ts

```diff
@@ -148,6 +161,37 @@ function writeLogsToFile(testName: string) {
     appendFileSync(testLogOutput, entries?.join('\n') ?? '', 'utf8')
 }
 
+/**
+ * Validates that emitted telemetry metrics 1. contain a result property and 2. contain a reason propery is
+ * the result is 'Failed'.
+ *
+ * TODO: While this catches cases in code that is tested, untested code will still release incomplete metrics.
+ * Consider using custom lint rules to address these cases if possible.
+ */
+function validateTelemetryMetrics() {
+    const failedStr: Result = 'Failed'
+    const telemetryRunDocsStr =
+        'Consider using `.run()` instead of `.emit()`, which will set these properties automatically. ' +
+        'See https://github.com/aws/aws-toolkit-vscode/blob/master/docs/telemetry.md#guidelines'
+
+    for (const m of globals.telemetry.logger.getAllMetrics()) {
+        if (m.Metadata) {
+            const metadata = mapMetadata([])(m.Metadata)
+            assert.ok(
+                metadata.result !== undefined,
+                `This test emits metric \`${m.MetricName}\` without the \`result\` property. ` +
```

Comment:
Nice. We can clarify a bit: we can drop reference to "This test" because the metric itself is the problem, not any test.
```suggestion
                `Metric \`${m.MetricName}\` was emitted without the \`result\` property. ` +
```